lighthouse-core/report/html/renderer/report-renderer.js

```diff
@@ -193,67 +196,62 @@ if (typeof module !== 'undefined' && module.exports) {
- *     displayValue: (string|Array<string|number>|undefined),
- *     helpText: string,
- *     scoreDisplayMode: string,
- *     error: boolean,
- *     score: (number|null),
- *     details: (!DetailsRenderer.DetailsJSON|undefined),
- * }}
+      rawValue: (number|boolean|undefined),
+      name: string,
+      description: string,
+      explanation?: string,
+      errorMessage?: string,
+      displayValue?: string|Array<string|number>,
+      helpText: string,
+      scoreDisplayMode: string,
+      error: boolean,
+      score: (number|null),
+      details?: DetailsJSON,
+  }} AuditResultJSON
  */
-ReportRenderer.AuditResultJSON; // eslint-disable-line no-unused-expressions
 
 /**
  * @typedef {{
- *     id: string,
- *     score: (number|null),
- *     weight: number,
- *     group: (string|undefined),
- *     result: ReportRenderer.AuditResultJSON
- * }}
+      id: string,
+      score: (number|null),
+      weight: number,
+      group?: string,
+      result: AuditResultJSON
+  }} AuditJSON
  */
-ReportRenderer.AuditJSON; // eslint-disable-line no-unused-expressions
 
 /**
  * @typedef {{
- *     title: string,
- *     id: string,
- *     score: (number|null),
- *     description: (string|undefined),
- *     manualDescription: string,
- *     auditRefs: !Array<!ReportRenderer.AuditJSON>
- * }}
+      title: string,
+      id: string,
+      score: (number|null),
+      description?: string,
+      manualDescription: string,
+      auditRefs: Array<AuditJSON>
+  }} CategoryJSON
  */
-ReportRenderer.CategoryJSON; // eslint-disable-line no-unused-expressions
 
 /**
  * @typedef {{
- *     title: string,
- *     description: (string|undefined),
- * }}
+      title: string,
+      description?: string,
+  }} GroupJSON
  */
-ReportRenderer.GroupJSON; // eslint-disable-line no-unused-expressions
 
 /**
  * @typedef {{
- *     lighthouseVersion: string,
- *     userAgent: string,
- *     fetchTime: string,
- *     timing: {total: number},
- *     requestedUrl: string,
- *     finalUrl: string,
- *     runWarnings: (!Array<string>|undefined),
- *     artifacts: {traces: {defaultPass: {traceEvents: !Array}}},
- *     audits: !Object<string, !ReportRenderer.AuditResultJSON>,
- *     categories: !Object<string, !ReportRenderer.CategoryJSON>,
- *     reportCategories: !Array<!ReportRenderer.CategoryJSON>,
- *     categoryGroups: !Object<string, !ReportRenderer.GroupJSON>,
- *     configSettings: !LH.Config.Settings,
- * }}
+      lighthouseVersion: string,
+      userAgent: string,
+      fetchTime: string,
+      timing: {total: number},
+      requestedUrl: string,
+      finalUrl: string,
+      runWarnings?: Array<string>,
+      artifacts: {traces: {defaultPass: {traceEvents: Array}}},
+      audits: Object<string, AuditResultJSON>,
+      categories: Object<string, CategoryJSON>,
+      reportCategories: Array<CategoryJSON>,
+      categoryGroups: Object<string, GroupJSON>,
+      configSettings: LH.Config.Settings,
+  }} ReportJSON
```

Comment:
it'd be nice to switch these to the `@property` syntax later on so we get some actual syntax highlighting :)

Reply:
> t'd be nice to switch these to the @property syntax later on

agreed, though I'm hoping the majority of the use of these will become references to the LHR types :) (it was purely laziness that I didn't convert these...I can if it would be better)